src/components/SearchResults/DataSearchResultItem.js

```diff
@@ -0,0 +1,156 @@
+    textDecoration: 'underline'
+  },
+  searchResultItem: {
+    color: theme.palette.primary.main,
+    fontWeight: 'bold'
+  },
+  resultType: {
+    fontWeight: 'bold'
+  }
+}));
+
+function InTopicDataResult({ topicId, item, title, visualType, chartId }) {
+  const classes = useStyles();
+  const [country, setCountry] = useState('');
+  const [sectionSlug, setSectionSlug] = useState('');
+  const [topicSlug, setTopicSlug] = useState('');
+
+  useEffect(() => {
+    fetch(
+      `${config.WP_BACKEND_URL}/wp-json/wp/v2/topic_page/${topicId}?_embed`
+    ).then(response => {
+      if (response.status === 200) {
+        response
+          .json()
+          .then(
+            ({ slug, acf: { section_topics: profileSection }, _embedded }) => {
+              setTopicSlug(slug);
+              if (profileSection && profileSection.length > 0) {
+                setSectionSlug(profileSection[0].post_name);
+              }
+              const category = _embedded['wp:term'][0][0];
+              setCountry(category);
+            }
+          );
+      }
+    });
+  }, [topicId]);
+
+  const link = `/profiles/${country.slug}/${sectionSlug}#${topicSlug}?indicator=indicator-${visualType}-${chartId}`;
+
+  return (
+    <div className={classes.root}>
+      <Typography variant="body1" className={classes.resultType}>
+        {item}
+      </Typography>
+      <Link href={link} as={link} className={classes.link}>
+        <Typography variant="body1" className={classes.searchResultItem}>
+          {country.name} - {title}
+        </Typography>
+      </Link>
+    </div>
+  );
+}
+
+InTopicDataResult.propTypes = {
+  topicId: PropTypes.number.isRequired,
+  visualType: PropTypes.string.isRequired,
+  chartId: PropTypes.number.isRequired,
+  title: PropTypes.string.isRequired,
+  item: PropTypes.string.isRequired
+};
+
+export const InTopicResult = InTopicDataResult;
+
+function InGeographyDataResult({ geoId, item, title, chartId }) {
+  const classes = useStyles();
+  const link = `/embed/${geoId}/${chartId}`; // TODO: link should be replaced with data-by-topics once we know section in belongs
+  const countryCode = geoId.split('-')[1].slice(2);
+  const country = config.countries.find(c => c.code === countryCode);
+
+  return (
+    <div className={classes.root}>
+      <Typography variant="body1" className={classes.resultType}>
+        {item}
+      </Typography>
+      <Link href={link} as={link} className={classes.link}>
+        <Typography variant="body1" className={classes.searchResultItem}>
+          {country.name} - {title}
+        </Typography>
+      </Link>
+    </div>
+  );
+}
+
+InGeographyDataResult.propTypes = {
+  chartId: PropTypes.number.isRequired,
+  geoId: PropTypes.number.isRequired,
+  title: PropTypes.string.isRequired,
+  item: PropTypes.string.isRequired
+};
+
+export const InGeographyResult = InGeographyDataResult;
+
+function DataSearchResultItem({ item, title, id, visualType, visualData }) {
```

Comment:
It could be I'm using outdated WP data but the `visualData` sometimes is just a html string and not JSON string. For example when I search for "Kenya", I get:
```
  item: 'Data',
  title: 'Kenyan President Uhuru Kenyatta',
  id: 550,
  visualData: '<p style="text-align: center;"><img class="size-medium wp-image-552 aligncenter" src="https://dashboard.takwimu.africa/wp-content/uploads/2019/12/6lX3Om0Yrouhuru-kenyatta-300x250.png" alt="" width="300" height="250" /></p>\r\n' +
    '<p style="text-align: center;">President Kenyatta is the son of Kenyaâ€™s first President Jomo Kenyatta (1964-78).</p>\r\n' +
    '<!--more-->\r\n' +
    '\r\n' +
    'He worked as a businessman in the agriculture and trade industries in the 1990s before entering politics, becoming a Minister for Local Government in 2001. He ran unsuccessfully for the presidency against Mwai Kibaki in 2002, before backing Kibaki in 2007 and joining his government, eventually serving as Deputy Prime Minister and holding various ministerial posts including Minister of Finance in the 2008 power-sharing government.\r\n' +
    '\r\n' +
    'Kenyatta ran successfully for the presidency in 2013, leading the Jubilee alliance in a slim victory over opposition candidate Raila Odinga. He won a second elected mandate in the 2017 elections, which were marred in controversy after Kenyattaâ€™s initial victory was overturned by the Supreme Court and the re-run was boycotted by the opposition.\r\n' +
    '\r\n' +
    'Historically, political power in Kenya has been heavily centralised, and though decentralisation reforms in 2010 have curtailed presidential powers, the presidency remains the most influential political office within the country. Nevertheless, opposition challenges to the legitimacy of government following the contested 2017 elections, the growing influence of county governments, and factional jostling within the Jubilee party serve to create competing political forces around Kenyattaâ€™s office.\r\n' +
    '\r\n' +
    'Kenyattaâ€™s Jubilee manifesto in both elections has largely focused around the theme of securing prosperity for all, reflected in significant investments in the countryâ€™s development, particularly in infrastructure and education. His 2017 manifesto covered the â€˜Big Fourâ€™ priorities, which aim to accelerate economic growth through investments in manufacturing, food security, health and housing. In order to deliver on this pledge, Kenyatta will need to strike a balance in spearheading investments and ensuring the sustainability of public finances amid rising public debt levels',
  visualType: ''
```
as one of the results which causes an error.

Reply:
@kilemensi I forgot about `carousel_topic` type. It should be handled together with analysis results and not data

Reply:
and `snippets`. But this will be address in another story